spec/controllers/tasks_controller_spec.rb

```diff
@@ -0,0 +1,392 @@
+require 'rails_helper'
+
+RSpec.describe TasksController, type: :controller do
+  let(:task) { FactoryGirl.create(:task) }
+  let(:another_task) { FactoryGirl.create(:task) }
+  let(:project) { task.project }
+  let(:user) { project.user }
+
+  describe "when suer is not authenticated" do
+    context "#index" do
+      it "redirects to sign in page" do
+        get :index, params: { project_id: project.id }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#show" do
+      it "redirects to sign in page" do
+        get :show, params: { project_id: task.project.id, id: task.id }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#new" do
+      it "redirects to sign in page" do
+        get :new, params: { project_id: project.id }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#edit" do
+      it "redirects to sign in page" do
+        get :edit, params: { project_id: task.project.id, id: task.id }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#update" do
+      it "redirects to sign in page" do
+        patch :update, params: { project_id: task.project.id, id: task.id, task: FactoryGirl.attributes_for(:task) }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#create" do
+      it "redirects to sign in page" do
+        post :create, params: { project_id: project.id, task: FactoryGirl.attributes_for(:task) }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#mark_as_done" do
+      it "redirects to sign in page" do
+        put :mark_as_done, params: { project_id: project.id, id: task.id, task: task.attributes }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context "#destroy" do
+      it "redirects to sign in page" do
+        delete :destroy, params: { project_id: project.id, id: task.id }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  context "when user is authenticated" do
+    before { sign_in(user) }
+
+    describe "#index" do
+      context "with proper params" do
+        before { get :index, params: { project_id: project.id } }
+
+        it "assigns tasks" do
```

Comment:
Imho this description is a little confusing. `#index` action doesn't assign tasks to someone. Maybe `passes @tasks to view` or something like that would be better.